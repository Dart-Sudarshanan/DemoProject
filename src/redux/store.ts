import { configureStore } from "@reduxjs/toolkit";
import authSlice from "./slices/authSlice";

const store = configureStore({
    reducer:{
        userAuth: authSlice.reducer
    }
});

export type ApplicationState = ReturnType<typeof store.getState >;

// export const selectUser = (state: ApplicationState) => state.userAuth;

export {store};