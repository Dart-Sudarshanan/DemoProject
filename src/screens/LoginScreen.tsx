import React from "react";
import { StyleSheet, View } from "react-native";
import UserForm from "../components/forms/UserForm";

function LoginScreen(): React.ReactElement {
  return (
    <View style={styles.container}>
      <UserForm />
    </View>
  );
}

export default LoginScreen;

const styles = StyleSheet.create({
  container: {
    justifyContent: "center",
    height: "100%",
    borderBottomColor: "#000",
  },
});
