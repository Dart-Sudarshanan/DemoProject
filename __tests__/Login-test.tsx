/**
 * @format
 */

import "react-native";
import React from "react";

// import Enzyme, { shallow } from "enzyme";
// import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

import { cleanup, fireEvent, render } from "@testing-library/react-native";

// Note: test renderer must be required after react-native.
import LoginScreen from "../src/screens/LoginScreen";
import { Provider } from "react-redux";
import { store } from "../src/redux/store";

describe("Login Form", () => {
  it("form rendered correctly", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <LoginScreen />
      </Provider>
    );
    const text = getByTestId("form-wrapper");
    expect(text).toBeTruthy();
  });

  it("Fields are empty", () => {
    const onSubmit = jest.fn();
    const { getByTestId, getAllByTestId, queryAllByText } = render(
      <Provider store={store}>
        <LoginScreen />
      </Provider>
    );

    const button = getByTestId("button");
    fireEvent.press(button);
    expect(queryAllByText("Both fields are required").length).toBe(1);
  });

  it("User is empty", () => {
    const onSubmit = jest.fn();
    const { getByTestId, getAllByTestId, queryAllByText } = render(
      <Provider store={store}>
        <LoginScreen />
      </Provider>
    );
    const input = getAllByTestId("login-input");
    expect(input).toHaveLength(2);
    fireEvent.changeText(input[1], "aa112");
    const button = getByTestId("button");
    fireEvent.press(button);
    expect(queryAllByText("Username is required").length).toBe(1);
  });

  it("Password is empty", () => {
    const onSubmit = jest.fn();
    const { getByTestId, getAllByTestId, queryAllByText } = render(
      <Provider store={store}>
        <LoginScreen />
      </Provider>
    );
    const input = getAllByTestId("login-input");
    expect(input).toHaveLength(2);
    fireEvent.changeText(input[0], "suda@gmail.com");
    const button = getByTestId("button");
    fireEvent.press(button);
    expect(queryAllByText("Password is required").length).toBe(1);
  });

  it("User not found", () => {
    const onSubmit = jest.fn();
    const { getByTestId, getAllByTestId, queryAllByText } = render(
      <Provider store={store}>
        <LoginScreen />
      </Provider>
    );

    const input = getAllByTestId("login-input");
    const button = getByTestId("button");
    expect(input).toHaveLength(2);
    fireEvent.changeText(input[0], "suda@gmail.com");
    fireEvent.changeText(input[1], "aa112");
    fireEvent.press(button);

    expect(queryAllByText("User not found").length).toBe(1);
  });

  it("Password incorrect", () => {
    const onSubmit = jest.fn();
    const navigationMock = jest.fn();
    const { getByTestId, getAllByTestId, queryAllByText } = render(
      <Provider store={store}>
        <LoginScreen navigation={{ push: navigationMock }} />
      </Provider>
    );

    const input = getAllByTestId("login-input");
    const button = getByTestId("button");
    expect(input).toHaveLength(2);
    fireEvent.changeText(input[0], "sudar@gmail.com");
    fireEvent.changeText(input[1], "aa112");
    fireEvent.press(button);

    expect(queryAllByText("Password incorrect").length).toBe(1);
  });

  it("Login Successful", () => {
    const onSubmit = jest.fn();
    const navigationMock = jest.fn();
    const { getByTestId, getAllByTestId, toJSON } = render(
      <Provider store={store}>
        <LoginScreen navigation={navigationMock} />
      </Provider>
    );

    const input = getAllByTestId("login-input");
    const button = getByTestId("button");
    expect(input).toHaveLength(2);
    fireEvent.changeText(input[0], "sudar@gmail.com");
    fireEvent.changeText(input[1], "aa1122");
    fireEvent.press(button);

    expect(navigationMock).toBeCalledWith("Home");
    // expect(toJSON()).toMatchSnapshot();
    // expect(queryAllByText("Password incorrect").length).toBe(1);
  });

  // it("it should render text in color #4d4949", () => {
  //   const { getByTestId, getByText, getAllByTestId, toJSON } = render(
  //     <Provider store={store}>
  //       <LoginScreen />
  //     </Provider>
  //   );
  //   const text = getByTestId("login-header");
  //   expect(text.props.style).toMatchObject({ color: "#4d4949", fontSize: 24, fontWeight: "700" });
  // });
});
