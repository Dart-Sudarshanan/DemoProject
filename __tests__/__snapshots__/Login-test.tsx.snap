// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form Login Successful 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "justifyContent": "center",
        "paddingHorizontal": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#4d4949",
          "fontSize": 24,
          "fontWeight": "700",
        }
      }
    >
      Login
    </Text>
    <View
      style={
        Object {
          "marginTop": 20,
        }
      }
      testID="form-wrapper"
    >
      <View
        style={
          Object {
            "margin": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "400",
            }
          }
        >
          Username
        </Text>
        <View>
          <TextInput
            input={
              Object {
                "name": "email",
                "onBlur": [Function],
                "onChange": [Function],
                "onDragStart": [Function],
                "onDrop": [Function],
                "onFocus": [Function],
                "value": "sudar@gmail.com",
              }
            }
            label="Username"
            meta={
              Object {
                "active": false,
                "asyncValidating": false,
                "autofilled": false,
                "dirty": true,
                "dispatch": [Function],
                "error": undefined,
                "form": "signIn",
                "initial": undefined,
                "invalid": false,
                "pristine": false,
                "submitFailed": false,
                "submitting": false,
                "touched": true,
                "valid": true,
                "visited": false,
                "warning": undefined,
              }
            }
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            style={
              Object {
                "borderColor": "#a39d9d",
                "borderRadius": 8,
                "borderWidth": 2,
                "padding": 10,
              }
            }
            testID="login-input"
            value="sudar@gmail.com"
          />
        </View>
      </View>
      <View
        style={
          Object {
            "margin": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "400",
            }
          }
        >
          Password
        </Text>
        <View>
          <TextInput
            input={
              Object {
                "name": "password",
                "onBlur": [Function],
                "onChange": [Function],
                "onDragStart": [Function],
                "onDrop": [Function],
                "onFocus": [Function],
                "value": "aa112",
              }
            }
            label="Password"
            meta={
              Object {
                "active": false,
                "asyncValidating": false,
                "autofilled": false,
                "dirty": true,
                "dispatch": [Function],
                "error": undefined,
                "form": "signIn",
                "initial": undefined,
                "invalid": false,
                "pristine": false,
                "submitFailed": false,
                "submitting": false,
                "touched": true,
                "valid": true,
                "visited": false,
                "warning": undefined,
              }
            }
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            secureTextEntry={true}
            style={
              Object {
                "borderColor": "#a39d9d",
                "borderRadius": 8,
                "borderWidth": 2,
                "padding": 10,
              }
            }
            testID="login-input"
            value="aa112"
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "margin": 10,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#4c51e6",
            "borderRadius": 4,
            "paddingHorizontal": 40,
            "paddingVertical": 10,
          }
        }
        testID="button"
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 18,
              "fontWeight": "500",
            }
          }
        >
          Log in
        </Text>
      </View>
    </View>
    <View>
      <Text>
        Password incorrect
      </Text>
    </View>
  </View>
</View>
`;
